src/tofromdatavalues.jl

```diff
@@ -52,17 +52,26 @@ struct IteratorRow{T}
     row::T
 end
 
+unwrap(::Type{T}, x) where {T} = convert(T, x)
+unwrap(::Type{Any}, x) = x.hasvalue ? x.value : missing
```

Comment:
Maybe replace this with `get(x, missing)`? That way you don't rely on the internal structure of `DataValue` (which might change in the future), and I think that should amount to the same code running in the end.